cms/djangoapps/contentstore/utils.py

```diff
@@ -1390,9 +1390,63 @@ def get_help_urls():
     return help_tokens
 
 
+def get_response_format(request):
+    return request.GET.get('format') or request.POST.get('format') or 'html'
+
+
+def request_response_format_is_json(request, response_format):
+    return response_format == 'json' or 'application/json' in request.META.get('HTTP_ACCEPT', 'application/json')
+
+
+def get_library_context(request, request_is_json=False):
+    """
+    Utils is used to get context of course home library tab.
+    It is used for both DRF and django views.
+    """
+    from cms.djangoapps.contentstore.views.course import (
+        get_allowed_organizations,
+        get_allowed_organizations_for_libraries,
+        user_can_create_organizations,
+        _accessible_libraries_iter,
+        _get_course_creator_status,
+        _format_library_for_view,
+    )
+    from cms.djangoapps.contentstore.views.library import (
+        LIBRARIES_ENABLED,
+    )
+
+    libraries = _accessible_libraries_iter(request.user) if LIBRARIES_ENABLED else []
```

Comment:
Does this return v1 + V2 libraries? I'm unfamiliar with how the courses api handles that.

Reply:
I am not sure. I believe it only returns v1 libraries. I haven't tested in a v2 context because we always like out to load the library mfe.

Reply:
yeah that is true, how will the homepage know about which version of librarys we are using?

Reply:
It is dependent on the library authoring waffle flag (`studio.library_authoring_mfe`).